This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tony Theodore <tonyt@logyst.com>
Date: Sat, 18 Nov 2017 22:47:03 +1100
Subject: [PATCH 1/1] allows use of ocaml tools for cross-compilation and use
 prefixed ocamldoc


diff --git a/Makefile b/Makefile
index 1111111..2222222 100644
--- a/Makefile
+++ b/Makefile
@@ -1,8 +1,12 @@
 # Makefile generated by OCamake 
 # http://tech.motion-twin.com
 .SUFFIXES : .ml .mli .cmo .cmx .cmi .mll .mly
+OCAMLC=@target@-ocamlc
+OCAMLOPT=@target@-ocamlopt
+OCAMLDOC=@target@-ocamldoc
 
-INSTALLDIR=`ocamlc -where`
+
+INSTALLDIR=@installdir@
 CFLAGS=
 LFLAGS= -a
 LIBS=
@@ -16,19 +20,19 @@ install: all opt
 
 doc:
 	mkdir doc
-	ocamldoc -sort -html -d doc xml.mli dtd.mli xmlParser.mli
+	$(OCAMLDOC) -sort -html -d doc xml.mli dtd.mli xmlParser.mli
 
 test.exe: xml-light.cma
-	ocamlc xml-light.cma test.ml -o test.exe
-	
+	$(OCAMLC) xml-light.cma test.ml -o test.exe
+
 test_opt.exe: xml-light.cmxa
-	ocamlopt xml-light.cmxa test.ml -o test_opt.exe
+	$(OCAMLOPT) xml-light.cmxa test.ml -o test_opt.exe
 
 xml-light.cma: xml_parser.cmo xml_lexer.cmo dtd.cmo xmlParser.cmo xml.cmo 
-	ocamlc -o xml-light.cma $(LFLAGS) $(LIBS) xml_parser.cmo xml_lexer.cmo dtd.cmo xmlParser.cmo xml.cmo
+	$(OCAMLC) -o xml-light.cma $(LFLAGS) $(LIBS) xml_parser.cmo xml_lexer.cmo dtd.cmo xmlParser.cmo xml.cmo
 
 xml-light.cmxa: xml_parser.cmx xml_lexer.cmx dtd.cmx xmlParser.cmx xml.cmx 
-	ocamlopt -o xml-light.cmxa $(LFLAGS) $(LIBS) xml_parser.cmx xml_lexer.cmx dtd.cmx xmlParser.cmx xml.cmx
+	$(OCAMLOPT) -o xml-light.cmxa $(LFLAGS) $(LIBS) xml_parser.cmx xml_lexer.cmx dtd.cmx xmlParser.cmx xml.cmx
 
 dtd.cmo: xml.cmi xml_lexer.cmi dtd.cmi
 
@@ -68,13 +72,13 @@ wclean:
 
 # SUFFIXES
 .ml.cmo:
-	ocamlc $(CFLAGS) -c $<
+	$(OCAMLC) $(CFLAGS) -c $<
 
 .ml.cmx:
-	ocamlopt $(CFLAGS) -c $<
+	$(OCAMLOPT) $(CFLAGS) -c $<
 
 .mli.cmi:
-	ocamlc $(CFLAGS) $<
+	$(OCAMLC) $(CFLAGS) $<
 
 .mll.ml:
 	ocamllex $<
